packages/postcss-reduce-initial/src/script/__tests__/io.js

```diff
@@ -0,0 +1,86 @@
+import nock from 'nock';
+import { handleError, toJSONString, write, generate } from '../lib/io.mjs';
+import testData from './sampleProperties.json';
+
+describe('toJSONString', () => {
+  const rawData = {
+    foo: 'bar',
+    'baz-qux': 'quux',
+    quuz: 'corge',
+    garply: 'waldo',
+    'fred-plugh': 'xyzzy-thud',
+  };
+
+  test('should have correct data type (string)', () => {
+    expect(typeof toJSONString(rawData)).toBe('string');
+  });
+
+  test('should produce parsable JSON', () => {
+    expect(JSON.parse(toJSONString(rawData))).toEqual(rawData);
+  });
+
+  test('should have a trailing newline', () => {
```

Comment:
I assumed it was a choice made by the author of the original script to comply with the POSIX standard and be compatible with tooling as you suggest. One could think that it's a bit of a moot point since it's not a file that should be edited directly, but since it's checked into source control the new line should be there for the sake of the diff tool at least

Reply:
Good point about version control. But it occurred to me that the eslint and prettier should take care of files ending with a new line, if that is not the case we should fix the linter config instead of adding this unit test. 
```
yarn eslint --print-config packages/postcss-reduce-initial/data/fromInitial.json
```
shows that eslint applies the `prettier/prettier` rules to the JSON files and it seems prettier doe ensure newlines  are present (it's even[ impossible to turn off](https://github.com/prettier/prettier/issues/2575))

Reply:
Yeah, that's also a valid point.

If I remove the `\n` and open the file in vscode + save it, then the prettier rule does take care of the newline. However if I just generate the file without the trialing `\n` and run `yarn lint` on the project it does not complain or fix it.

Do you have some extra linting beyond that set up in CI that would take care of this? if not something probably needs to be changed to make it happen automagically

Reply:
There's no extra rules in CI.  Are you sure that saving the file in VsCode adds a newline because of the project linter settings? It might be a VScode setting. It seems that the eslint config needs some improvements.

Reply:
Running `yarn prettier --write packages/postcss-reduce-initial/data/fromInitial.json ` _removes_ the newline at the end of the file. I'd say definitely do not test for the presence of a newline, at this point it's not clear whether it makes any difference/which is better.

Reply:
Strange that you are seeing that behaviour from prettier - it's the opposite here with the same command. And they also mention that "files always end with a single newline" in their official "rationale" [here](https://prettier.io/docs/en/rationale.html#empty-lines)

I agree that this is something that linting should address in general, to catch cases where it is forgotten/overlooked.

But I'm leaning towards it being appropriate in principle to follow the POSIX standard when writing the file. So that it is validly formatted for most operating systems/tools upon creation - and not relying on external tools such as a linter to update it at some point after. And in that case also leave the test, but update the name so that it documents better why the newline is there.

Reply:
I see it this way: suppose that one day the acquire script does not write a new line anymore. The cssnano plugin would still work (because in fact it does work without a newline), but the test would break. In that case the test failure would be a false signal, since the program in fact would still be working fine. Therefore the test failure gives no information, so the test is redundant.